llvm/include/llvm/Analysis/ScalarEvolution.h

```diff
@@ -65,6 +65,117 @@ enum SCEVTypes : unsigned short;
 
 extern bool VerifySCEV;
 
+class SCEV;
+
+class SCEVUse : public PointerIntPair<const SCEV *, 3> {
```

Comment:
I think you need two bits: one for NUW and one for NSW. The canonical bit can be dropped, if you can canonicalize on construction.